### PR TITLE
SOC-1291 Use of contributors component and avatar component

### PR DESCRIPTION
**Pull request by armonr**
Targeting dev branch + minor changes from #1424 

* Use the user avatar component in discussion pages using raw html
* Addition of Discussion contributors component to the hero unit
* Addition of Discussion contributors component to left rail + post details page mobile

/CC @mklucsarits @garthwebb @bkoval 

https://wikia-inc.atlassian.net/browse/SOC-1291

### CODE REVIEW

**Comment by mklucsarits** [comment]
LGTM

File: front/scripts/main/models/DiscussionForumModel.ts [hunk 1]
```diff
@@ -63,10 +64,24 @@ App.DiscussionForumModel.reopenClass({
...
 
+					posts.forEach(function (post: any) {
+						var author: any;
+						if (post.hasOwnProperty('createdBy')) {
+							author = post.createdBy;
+							author.url = M.buildUrl({
+								namespace: 'User',
+								title: author.name
+							});
+							contributors.push(author)
```
Comment (rogatty) [inline front/scripts/main/models/DiscussionForumModel.ts]: Missing semicolon, it surely causes warnings in console.

File: front/scripts/main/models/DiscussionPostModel.ts [hunk 2]
```diff
@@ -17,7 +18,7 @@ App.DiscussionPostModel = Em.Object.extend({
 				url: 'https://' + M.prop('servicesDomain') + '/discussion/' +
 					 this.wikiId + '/threads/' + this.threadId +
 					 '?responseGroup=full' +
-					 '&sortDirection=descending' +
+					 '&sortDirection=descending&sortKey=creation_date' +
```
Comment (rogatty) [inline front/scripts/main/models/DiscussionPostModel.ts]: Seems like convention here is to keep one param per line.

Comment (bkoval) [inline front/scripts/main/models/DiscussionPostModel.ts]: https://github.com/Wikia/mercury/pull/1469/files#diff-6ca9940dddda8e8ce9d94b5f94033310R59 - not really :)

@rogatty I created a ticket for abstracting and improving the way we're building discussion urls (https://wikia-inc.atlassian.net/browse/SOC-1336), hopefully we'll be able to implement it soon.

File: front/styles/main/component/_discussion-contributors.scss [hunk 3]
```diff
@@ -26,8 +26,30 @@ $discussion-contributors-avatar-dimension: 30px;
...
+}
+
+.rail {
+	.discussion-contributors {
+		display: inline-block;
+		margin-right: 30px;
+	}
+
+	.discussion-contributors-title {
+		color: $oil;
```
Comment (bkoval) [inline front/styles/main/component/_discussion-contributors.scss]: https://github.com/Wikia/mercury/pull/1470#discussion_r40916653 :)

File: front/templates/main/components/post-detail.hbs [hunk 4]
```diff
@@ -1,10 +1,10 @@
...
-		<a href="{{authorUrl}}">
-			<img src="{{post.createdBy.avatarUrl}}">
-		</a>
-	</figure>
-	<a href="{{authorUrl}}" class="author">{{post.createdBy.name}}</a>
+	{{user-avatar
+		avatarUrl=author.avatarUrl
+		profileUrl=authorUrl
+		username=author.name
+		displayUsername=true
```
Comment (bkoval) [inline front/templates/main/components/post-detail.hbs]: I think the convention is to pass properties to a component in an alphabetical order :)

File: front/templates/main/components/post-reply.hbs [hunk 5]
```diff
@@ -6,14 +6,12 @@
...
-			</a>
-		</figure>
-		<a href="{{authorUrl}}" class="author">
-			{{post.createdBy.name}}
-		</a>
+		{{user-avatar
+			avatarUrl=author.avatarUrl
+			profileUrl=authorUrl
+			username=author.name
+			displayUsername=true
```
Comment (bkoval) [inline front/templates/main/components/post-reply.hbs]: https://github.com/Wikia/mercury/pull/1469/files#r40922205

**Comment by bkoval** [comment]
LGTM :+1:

**Comment by armonr** [comment]
Thanks @bkoval 
Going to merge this...

**Pull request merged**